src/clover/mains_supply/grid.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/python3
+########################################################################################
+# grid.py - Grid-profile generation module.                                            #
+#                                                                                      #
+# Author: Phil Sandwell                                                                #
+# Copyright: Phil Sandwell, 2018                                                       #
+# License: Open source                                                                 #
+# Most recent update: 14/07/2021                                                       #
+#                                                                                      #
+# For more information, please email:                                                  #
+#     philip.sandwell@gmail.com                                                        #
+########################################################################################
+"""
+grid.py - The grid-generation module for CLOVER.
+
+This module generates grid-availability profiles for CLOVER.
+
+"""
+
+from logging import Logger
+from typing import Dict
+
+import pandas as pd  # pylint: disable=import-error
+
+from .__utils__ import get_intermittent_supply_status
+
+__all__ = (
+    "get_lifetime_grid_status",
+    # "GridStatusThread",
+)
+
+
+def get_lifetime_grid_status(
+    generation_directory: str, grid_times: pd.DataFrame, logger: Logger, max_years: int
+) -> Dict[str, pd.DataFrame]:
+    """
+    Calculates, and saves, the grid-availability profiles of all input types.
+
+    Inputs:
+        - generation_directory:
+            The directory in which auto-generated files should be saved.
+        - grid_times:
+            Grid inputs information, read from the grid-inputs file.
+        - logger:
+            The logger to use for the run.
+        - max_years:
+            The maximum number of years for which the simulation should run.
+
+    Outputs:
+        - grid_profiles:
+            A dictionary mapping the grid name to the grid profile.
+
+    """
+
+    return get_intermittent_supply_status(
+        generation_directory, "grid", logger, max_years, grid_times
+    )
+
+
```

Comment:
What shall we do with the below lines? The `change_grid_coverage` function could be useful, but is quite niche

Reply:
I'll delete my random nonsense 😛 , but I'm not sure what we want to do about the other functions. We could integrate something on the command-line maybe??

Reply:
Make a separate entry script.